scripts/examples/smapiv2.py

```diff
@@ -1,19 +1,22 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from __future__ import print_function
-import os, sys, time, socket, traceback
+import os, sys, io, time, socket, traceback
 
-log_f = os.fdopen(os.dup(sys.stdout.fileno()), "aw")
+log_f = io.open(os.dup(sys.stdout.fileno()), "w")
```

Comment:
Opening without the append mode may cause the output file (descriptor in this case) to be truncated. The 1:1 conversion would be to use `"a"` everywhere where `"aw"` had do be used with Python2.

The explanation is here:
https://stackoverflow.com/questions/42824474/python-open-built-in-function-difference-between-modes-aw-and-a-or-aw-and
```suggestion
log_f = io.open(os.dup(sys.stdout.fileno()), "a")
```

Reply:
@bernhardkaindl with append mode ,we will get illegal seek error.

Reply:
@acefei Let me go through the link and do necessary changes.

Reply:
Oops, I overlooked that this is just accessing stdout, not an actual file.
In this case, accessing stdout with "w" just works, as there is no "truncating" of stdout.

On @acefei's comment regarding `with`:

Normally, when opening files, the recommendation to use `with` to open files is sound.

But in these cases, an already opened stdout file descriptor is just reused, it's not a real `open()`.

In the worst case, `with` may close stdout, and if you end up really closing stdout, you terminate your ability to send any output to stdout easily.

So @acefei recommendation/comment only applies to regular open(), not when just accessing a file descriptor that is already opened like here with `.fileno()`.